File: PostBox/src/main/java/com/mvp/model/BoxModel.java
package com.mvp.model;

import com.mvp.bean.BaseBox;
import com.mvp.bean.Box3;
import com.mvp.bean.Box5;

public class BoxModel implements IBoxModel{

    private BaseBox[] boxes = new BaseBox[2];

    @Override
    public void setBoxSize(float length, float width, float height) {
        boxes[0] = new Box3(length, width, height);
        boxes[1] = new Box5(length, width, height);
    }

    @Override
    public String throwVerifiedBoxBackToClient() {
        System.out.println("Model : verified input box size by Model");
        for (BaseBox box : boxes) {
            if (box instanceof Box3) {
                if (((Box3) box).validate()) {
                    return "box3";
                } else {
                    continue;
                }
            } else if (box instanceof Box5) {
                if (((Box5) box).validate()) {
                    return "box5";
                } else {
                    return null;
                }
            }
        }
        return null;
    }

}

File: PostBox/src/main/kotlin/com/mvp/model/BoxModel.kt
package com.mvp.model.kt

import com.mvp.bean.kt.BaseBox
import com.mvp.bean.kt.Box3
import com.mvp.bean.kt.Box5

class BoxModel : IBoxModel {

    private lateinit var box3 : BaseBox
    private lateinit var box5: BaseBox

    override fun setBoxSize(length: Float?, width: Float?, height: Float?) {
        box3 = Box3(length, width, height)
        box5 = Box5(length, width, height)
    }

    override fun returnVerifiedBoxToClient(): String? {
        println("Model : verified input box size by Model")
        if (box3.validateReceivedBox(box3.lengthOfBox3, box3.widthOfBox3, box3.heightOfBox3)) {
            return "box3"
        } else if (box5.validateReceivedBox(box5.lengthOfBox5, box5.widthOfBox5, box5.heightOfBox5)) {
            return "box5"
        } else {
            return null
        }
    }
}